atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CellValuePutter.java

```diff
@@ -100,7 +123,12 @@ public Void apply(CassandraClient client) throws Exception {
                             MutationMap map = new MutationMap();
                             for (Map.Entry<Cell, Value> e : partition) {
                                 Cell cell = e.getKey();
-                                Column col = CassandraKeyValueServices.createColumn(cell, e.getValue());
+                                Column col = overwriteTimestamps
+                                        ? CassandraKeyValueServices.createColumn(cell, e.getValue())
+                                        : CassandraKeyValueServices.createColumn(
+                                                cell,
+                                                e.getValue(),
+                                                mutationTimestampGetter.apply(e.getValue().getTimestamp()));
```

Comment:
expecting this code to go away, but again, yeah this gets called once per cell.

Reply:
(Now only called once per API call - interface mentions that this should be cheap as well)